doc/design/audit-trail-mvp.md

```diff
@@ -0,0 +1,156 @@
+# Audit Trail MVP
+
+NIH has required that DANDI Archive generate audit logs while leaving undefined
+what that means. We are choosing to fulfill the requirement with a feature that
+indicates to Dandiset owners and admins the sequence of significant actions
+taken to modify Dandisets, along with identification of who performed each
+action.
+
+The major motivation is to inform owners about how their Dandisets have changed
+when a given owner did not make the change and wishes to know who to speak to
+about those changes. In essence, it is a tool to support effective
+collaboration.
+
+## Requirements
+
+1. **Persistent tracking of significant actions.** Whenever a *significant
+   action* (see list [below](#significant-actions)) occurs successfully, the archive will generate a
+   persistent record of that action, including the following data:
+    - The date and time
+    - The authorized user who triggered the action (note that this may be an actual user or the system itself for internal automated actions)
+    - The ID of the Dandiset on which the action occurs
+    - Event-specific data about the action itself
+
+2. **Display of audit log (to owners and admins only).** The Dandiset landing
+   page will display a rendered (i.e. human-readable) version of the log event
+   data (as long as the viewing user is an owner of the Dandiset, or an admin).
+
+   A possible example of such a display follows (the final rendered form may
+   differ from this depending on best practices, etc.):
+    
+    ```
+    YYYYMMDDTHHMMSS.mmmm: X added asset at path (/full/path) checksum
+    YYYYMMDDTHHMMSS.mmmm: X updated asset at path (/full/path) checksum
+    YYYYMMDDTHHMMSS.mmmm: X removed asset at path (/full/path) checksum
+    YYYYMMDDTHHMMSS.mmmm: X updated dandiset metadata
+    ```
+    
+3. **Extensible significant action list.** The significant actions the system is
+   capable of tracking should be extensible--that is, in the future new actions
+   should be able to be added to the system, and existing actions should be able
+   to expand the specific data they carry.
+
+## Significant Actions
+
+The following is the list of actions performed on Dandisets that are considered
+"significant" (i.e., require an audit record to be generated), along with any
+additional, event-specific data each will carry with it.
+
+|  | General data | Specific data |
+| --- | --- | --- |
+| Create Dandiset | date_time, user_id, dandiset_id | metadata |
+| Add owner | date_time, user_id, dandiset_id | owner_id |
+| Remove owner | date_time, user_id, dandiset_id | owner_id |
+| Update Dandiset metadata | date_time, user_id, dandiset_id | metadata |
```

Comment:
if for 
![image](https://github.com/dandi/dandi-archive/assets/39889/dd50e2fd-99c7-490a-8153-27cf395fd70c)
should we also get `version` into "specific data"?

```suggestion
| Update Dandiset metadata | date_time, user_id, dandiset_id | version, metadata |
```

Reply:
If you look at https://github.com/dandi/dandi-archive/blob/880a1022c590e4ef6606e0f6a00e16a98e90da9b/dandiapi/api/views/version.py#L90-L95 you'll see that only `draft` versions can have their metadata updated, hence no need to record the version.

Reply:
ATM - yes. But I thought we had already cases where we - admins - either wanted or did update metadata.  We might have not gone through this endpoint though.  We have issues for updates of metadata on assets (https://github.com/dandi/dandi-archive/issues/1450) and discussed in the past aspects of metadata migration (https://github.com/dandi/dandi-archive/issues/565)...

But I guess we could keep as is, and get to change schema as we get to that point of need.

Reply:
I would say let's keep this as-is for now. I'm pretty sure we didn't go through this endpoint to update metadata on published versions (since, as I showed above, that simply wouldn't have worked), and if we want to record a special audit event for metadata migration, we can add that in as part of the work on #565.